src/ride_type.cpp

```diff
@@ -775,8 +780,10 @@ void RidesManager::Save(Saver &svr)
 	}
 	svr.PutWord(count);
 	for (uint16 i = 0; i < count; i++) {
-		if (this->instances[i] == nullptr || this->instances[i]->state == RIS_ALLOCATED) continue;
-		this->instances[allocated_ride_indexes[i]]->Save(svr);
+		RideInstance *r = this->instances[allocated_ride_indexes[i]];
+		svr.PutByte(static_cast<uint8>(r->GetKind()));
```

Comment:
Any reason that these two lines were moved outside of `RideInstance::Save`?

Reply:
These two pieces of information are needed by the `RideManager` on loading before it can decide which subclass of `RideInstance` it needs to instantiate. The derived classes save their own version number before calling the base class's Save function though. 
If we kept the old order, the data which is required to instantiate the correct class would be hidden behind several subclass version numbers, which can only be peeled away after the class has been instantiated. So we need to move this data where it can be loaded prior to figuring out which type of ride to allocate.
I hope my explanation is clear… :)

Reply:
Makes sense. Thank you :)